archaius2-api/src/main/java/com/netflix/archaius/api/config/LayeredConfig.java

```diff
@@ -0,0 +1,47 @@
+package com.netflix.archaius.api.config;
+
+import com.netflix.archaius.api.Config;
+import com.netflix.archaius.api.Layer;
+import com.netflix.archaius.api.exceptions.ConfigException;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Composite Config where the override order is driven by Layer keys.
+ */
+public interface LayeredConfig extends Config {
```

Comment:
Long-term what happens to the Config interface? Would we even want to keep it? Will we continue to expose bindings for it? (beyond whatever is needed for backwards compatibility)

Reply:
Yes, this is mostly for backwards compatibility.  Long term I'd like to get rid of Config in favor of a PropertyResolver and PropertySource, similar to what Spring has.  This should ease the migration in the future as well.  Once this has been released I'll be making more changes towards introducing PropertyResolver and PropertySource derived interfaces.